variables.tf

```diff
@@ -1,3 +1,14 @@
+variable "gitlab_project_name" {
+  description = "The name of the Gitlab project that hosts the Gitlab Agent configuration. If not provided, the module will use the project defined in `gitlab_project_path_with_namespace`."
+  type        = string
+  default     = ""
+}
+
+variable "gitlab_root_namespace" {
+  description = "The full path of the root namespace in Gitlab. This is the namespace where the Gitlab Agent will be granted access."
+  type        = string
+}
```

Comment:
Is it possible to make this variable optional?

I think we can use the previous logic to determine the root namespace when the `gitlab_project_name` is not empty. What do you think?

Reply:
We always have the root namespace (is the key of the map we pass to the module), mapping it to a variable makes the code simplier and more readable.

The previuos logic relies on the project being already created so we can use the project name to find out the root namespace. When `gitlab_project_name` is not empty we don't have any means to find out the root namespace as the project doesn't exist yet.